android/sdk/src/main/java/com/taobao/weex/ui/component/list/DragSupportCallback.java

```diff
@@ -0,0 +1,332 @@
+ *
+ *    APPENDIX: How to apply the Apache License to your work.
+ *
+ *       To apply the Apache License to your work, attach the following
+ *       boilerplate notice, with the fields enclosed by brackets "[]"
+ *       replaced with your own identifying information. (Don't include
+ *       the brackets!)  The text should be enclosed in the appropriate
+ *       comment syntax for the file format. We also recommend that a
+ *       file or class name and description of purpose be included on the
+ *       same "printed page" as the copyright notice for easier
+ *       identification within third-party archives.
+ *
+ *    Copyright 2016 Alibaba Group
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.taobao.weex.ui.component.list;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.StaggeredGridLayoutManager;
+import android.support.v7.widget.helper.ItemTouchHelper;
+
+import com.taobao.weex.ui.view.listview.adapter.ListBaseViewHolder;
+import com.taobao.weex.utils.WXLogUtils;
+
+/**
+ * Description:
+ * An implementation of {@link ItemTouchHelper.Callback}. with drag&drop support:)
+ *
+ * <p>
+ * Created by rowandjj(chuyi)<br/>
+ * Date: 17/4/22<br/>
+ * Time: 上午11:25<br/>
+ */
+class DragSupportCallback extends ItemTouchHelper.Callback{
+    private final WXDragHelper mWXDragHelper;
+    private boolean enableDifferentViewTypeDrag = false;
+
+    private int dragFrom = -1;
+    private int dragTo = -1;
+
+    private static final String TAG = "WXListExComponent";
+
+    DragSupportCallback(@NonNull WXDragHelper WXDragHelper) {
+        this.mWXDragHelper = WXDragHelper;
+        this.enableDifferentViewTypeDrag = false;
+    }
+
+    DragSupportCallback(@NonNull WXDragHelper WXDragHelper, boolean enableDifferentViewTypeDrag) {
+        this.mWXDragHelper = WXDragHelper;
+        this.enableDifferentViewTypeDrag = enableDifferentViewTypeDrag;
+    }
+
+    @Override
+    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
+        if(recyclerView.getLayoutManager() instanceof GridLayoutManager || recyclerView.getLayoutManager() instanceof StaggeredGridLayoutManager){
+            final int dragFlags = ItemTouchHelper.UP | ItemTouchHelper.DOWN | ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT;
```

Comment:
Just pass the expression to the method, the final variable `dragFlags` is useless.